home task 9+10

msantipova opened the pull request:
(9)save response in ArrayResponse
(9)add operation change status
(10)add exception handling

msantipova committed:
home task 9+10
(9)save response in ArrayResponse
(9)add operation change status
(10)add exception handling

VasilevPoul commented on pom.xml:
```diff
@@ -13,8 +13,8 @@
-                    <source>6</source>
-                    <target>6</target>
+                    <source>7</source>
+                    <target>7</target>
```
16 и 17 строка, для использования лямбда выражений(это те которые в assertAll у тебя) надо задать 8 и более

VasilevPoul commented on src/main/java/com/tinkoff/edu/app/ArrayResponse.java:
```diff
@@ -0,0 +1,50 @@
+            if (responses[counter].getRequestId().equals(id)) {
+                break;
+            }
+        }
```
Данная форма цикла for устарела, и если тебе не нужен индекс то используй следующий формат:
```
        for (LoanResponse response : responses) {
            if (response.getRequestId().equals(id)) {
                break;
            }
        }
```

Проверь по всему проекту

VasilevPoul replied:
Вцелом в этой функции тебе надо прямо в цикле делать return вместо break, а после тела цикла можно прокидывать ошибку т.к. за весь цикл не удалось обнаружить записи с нужным UUID

VasilevPoul commented on src/main/java/com/tinkoff/edu/app/ArrayResponse.java:
```diff
@@ -0,0 +1,50 @@
+                break;
+            }
+        }
+        System.out.println( "ArrayResponse {" + responses[counter].getRequestId()
```
Используй printf вместо println
` System.out.println("value: %s and %s", 500, "AAA");`
напишет в консоль:
`value: 500 and AAA`

а при формировании строки всегда вместо '+' используй:
`String.format("value: %s and %s", 500, "AAA")`

VasilevPoul commented on src/main/java/com/tinkoff/edu/app/LoanResponse.java:
```diff
@@ -1,24 +1,52 @@
+        if (o == null || getClass() != o.getClass()) return false;
+        LoanResponse that = (LoanResponse) o;
+        return Objects.equals(request, that.request) &&
+                responseType == that.responseType &&
```
Почему ты в 44 строке не используешь Objects.equals ?

msantipova committed:
home task 9+10
(9)save response in ArrayResponse
(9)add operation change status
(10)add exception handling

msantipova commented:
исправлено
